[inductor] skip non-trivial tiling if unbacked symints are present

ColinPeppler opened the pull request:
Take two of https://github.com/pytorch/pytorch/pull/149994.

This time we just skip `convert_tiling_to_3d` and `candidate_tilings` if there exists unbacked symints.

Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #150225



cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @ipiszy @chenyang78 @kadeng @muchulee8 @amjames @chauhang @aakhundov

ColinPeppler committed:
[inductor] make non-trivial unbacked symint aware

[ghstack-poisoned]

ColinPeppler committed:
Update on "[inductor] make non-trivial unbacked symint aware"

Take two of https://github.com/pytorch/pytorch/pull/149994.

This time we just skip `convert_tiling_to_3d` and `candidate_tilings` if there exists unbacked symints.




cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy chenyang78 kadeng muchulee8 amjames chauhang aakhundov

[ghstack-poisoned]

eellison reviewed (approved):
SGTM - i think the other pr is

ColinPeppler committed:
Update on "[inductor] skip non-trivial tiling if unbacked symints are present"

Take two of https://github.com/pytorch/pytorch/pull/149994.

This time we just skip `convert_tiling_to_3d` and `candidate_tilings` if there exists unbacked symints.




cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy chenyang78 kadeng muchulee8 amjames chauhang aakhundov

[ghstack-poisoned]

ColinPeppler commented:
@pytorchbot merge
